src/cmd/sql_cmd_test.cc

```diff
@@ -3286,6 +3286,53 @@ TEST_P(DBSDKTest, CreateIfNotExists) {
     ASSERT_TRUE(cs->GetNsClient()->DropDatabase("test2", msg)) << msg;
 }
 
+TEST_P(DBSDKTest, MapTypeTable) {
+    auto cli = GetParam();
+    cs = cli->cs;
+    sr = cli->sr;
+    absl::BitGen gen;
+    auto db = absl::StrCat("db_", absl::Uniform(gen, 0, std::numeric_limits<int32_t>::max()));
+    auto table = absl::StrCat("tb_", absl::Uniform(gen, 0, std::numeric_limits<int32_t>::max()));
+
+    ProcessSQLs(sr, {
+                        "set session execute_mode = 'online'",
+                        absl::StrCat("create database ", db),
+                        absl::StrCat("use ", db),
+                        absl::Substitute("create table $0 (id string, val map<int, string>)", table),
+                        absl::Substitute("insert into $0 values ('1', map(12, '23')) ", table),
```

Comment:
how to insert map fields with placeholder

Reply:
not possible util codegen insert row support parameters

Reply:
just user interface. and one sql that contains map  placeholder only execute codegen at the first run time?

Reply:
it will codegen a function with extra function parameter, this parameter type might be `codec::Row` or `node::ExprListNode`, I'd prefer design it later